Barceloneta based Classic UI theme

petschki opened the pull request.

petschki committed:
barceloneta based theme doc

petschki committed:
remove notes

petschki committed:
variables and typo

gogobd commented:
Hi, @petschki !

* Maybe consider mentioning the version of npm you would recommend and that the generated theme addon needs to be installen (via Plone Panel).

Please mention that the file that should contain the font definitions is `_variables.scss`. I used a font other than the default font just to be able to see the change immediately:

```
$font-family-sans-serif:                    "Papyrus", "Fantasy";
$font-family-condensed:                     "Papyrus", "Fantasy"; //just on toolbar
$font-family-serif:                         "Papyrus", "Fantasy";
```

* This `$enabled-rounded: false;` switch didn't work.

* I was assuming that the changes mentioned in "Maps" would go into `maps.scss`, it's not mentioned but at the same time quite obvious i.e. most likely everyone's first guess.

petschki committed:
improvements

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-barceloneta

stevepiercy commented:
I just did a merge commit to bring it inline with `main` due to the failing docs test. We will see if it passes now.

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+```
+
+Give your theme a name and optionally commit the changes.
+After that the theming structure is setup and ready for customization.
````
```suggestion
After that the theming structure is set up and ready for customization.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
-### `base.scss`
+All the theming relevant files are now located inside `src/plonetheme/themebasedonbarceloneta/theme/`:
+
+```
````
````suggestion
```text
````

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+    ├── card-group.html
+    └── list.html
+```
+
````
Let's convert this into a more compact definition list, instead of a bunch of headings with sections, which will also make it more a part of the current section. If a definition has more than one sentence, then indent them to align with the start of the first sentence. Example:

```
`index.html`
:   Basic HTML structure for the site layout.

`manifest.cfg`
:   Basic theme configuration for the backend.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+plonecli create addon plonetheme.themebasedonbarceloneta
+```
+
+Then change the working directory into the created package and add the
````
Please use one sentence per line. Docs are not code. This makes it easier to review and make suggestions, if needed, and avoids the need to rewrap if it gets edited.

```suggestion
Then change the working directory into the created package and add the basic theme structure with the following command.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+```
+
+Then change the working directory into the created package and add the
+basic theme structure with:
````
```suggestion
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+
+### `rules.xml`
+
+-   Diazo rules which translate the `index.html` and fills it with content
```
```suggestion
-   Diazo rules which translate the `index.html` and fill it with content from the backend.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+### `rules.xml`
+
+-   Diazo rules which translate the `index.html` and fills it with content
+    from the backend.
```
```suggestion
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+### `scss/_variables.scss`
 
--   Basics required for backend
+-   Override Bootstrap and/or Barceloneta variables here.
```
```suggestion
-   Override Bootstrap and Barceloneta variables here.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+### `scss/theme.scss`
 
--   Barceloneta theme basics
+-   Base theme file which follows "Option B" of customizing Bootstrap imports
```
```suggestion
-   Base theme file which follows "Option B" of customizing Bootstrap imports to enable injections of custom variables and maps.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
 
--   Barceloneta theme basics
+-   Base theme file which follows "Option B" of customizing Bootstrap imports
+    to enable custom variable and map injections. Note that the order of these imports is mandatory to ensure overriding the defaults. See https://getbootstrap.com/docs/5.3/customize/sass/#importing
```
```suggestion
    Note that the order of these imports is mandatory to ensure overriding the defaults.
    See https://getbootstrap.com/docs/5.3/customize/sass/#importing.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
 
--   npm package with files required for Plone
--   dependencies required by theme to compile
+To compile the SCSS code you have to install Node.js (v16+) and the required dependencies with `npm`. Then run the package script `build` inside the `theme/` folder:
```
One sentence per line.
```suggestion
To compile the SCSS code you have to install Node.js (v16+) and the required dependencies with `npm`.
Then run the package script `build` inside the `theme/` folder:
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+```
+
+this compiles the SCSS resources on the fly when you change something inside
+the `scss/` folder.
````
Does this also allow the developer to preview their changes in the browser? If so, please mention it.

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+npm install
+npm run build
+```
+
````
```suggestion


(classic-ui-theming-barceloneta-compiling-theme-resources-label)=

## Compiling theme resources

```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+As a convenience `bobtemplates.plone` has created three files to customize
+variables, maps and custom SCSS code.
+
+```
````
````suggestion
```text
````

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+scss/_variables.scss
+```
+
+### SCSS and root variables
````
```suggestion

(classic-ui-theming-barceloneta-scss-and-root-variables-label)=

### SCSS and root variables
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+
+### SCSS and root variables
+
+To set a custom font you define the font variables in `scss/_variables.scss`:
```
```suggestion
To set a custom font, you define the font variables in `scss/_variables.scss`:
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+$font-family-serif: Georgia, Norasi, serif;
+```
+
+This will override the default values from barceloneta.
````
```suggestion
This will override the default values from Barceloneta.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
 
--   Default components are extended with some custom components, such as a select or dropdown menu.
--   We use Boostrap variables in these cases.
+Same procedure if you want to disable rounded corners for borders:
```
```suggestion
The following example shows how to disable rounded corners for borders.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+
+
+
+### Maps
```
```suggestion


(classic-ui-theming-barceloneta-maps-label)=

### Maps
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+
+### Maps
+
+Maps are key value pairs to make CSS generation easier. As an example this is the workflow colors map:
```
```suggestion
Maps are key/value pairs to make CSS generation easier.
As an example, the following example shows the workflow colors map:
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+) !default;
+```
+
+If you have a custom workflow state you can add your state color to the default map. Just add the following to `scss/_maps.scss`:
````
```suggestion
If you have a custom workflow state, you can add your state color to the default map in `scss/_maps.scss` as shown below.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+```
+
+
+### Custom CSS code
````
```suggestion

(classic-ui-theming-barceloneta-custom-css-code-label)=

### Custom CSS code
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+
+### Custom CSS code
+
+Inside the file `theme/_custom.scss` you can write all your custom CSS/Sass code
```
```suggestion
Inside the file `theme/_custom.scss`, you can write all your custom CSS/Sass code to adapt the theme to your needs.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+### Custom CSS code
+
+Inside the file `theme/_custom.scss` you can write all your custom CSS/Sass code
+to adapt the theme to your needs. Feel free to add more files inside the `scss/` folder to make your code more readable. Don't forget to import your custom files in `scss/theme.scss`.
```
```suggestion
Feel free to add more files inside the `scss/` folder to make your code more readable.
Don't forget to import your custom files in `scss/theme.scss`.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
+to adapt the theme to your needs. Feel free to add more files inside the `scss/` folder to make your code more readable. Don't forget to import your custom files in `scss/theme.scss`.
+
+
+(classic-ui-theming-barceloneta-default-variables-properties-label)=
```
```suggestion
(classic-ui-theming-barceloneta-default-variables-and-properties-label)=
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
````diff
@@ -11,59 +11,330 @@ myst:
+$enabled-rounded: false;
+```
+
+A complete list of all properties see {ref}`classic-ui-theming-barceloneta-default-variables-properties-label`.
````
```suggestion
A complete list of all properties see {ref}`classic-ui-theming-barceloneta-default-variables-and-properties-label`.
```

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,330 @@ myst:
 
 
-## Bootstrap components
+### Properties
```
To avoid duplicate heading at the end of the file.

```suggestion
### SCSS and properties
```

stevepiercy reviewed (changes requested):
Mostly a bunch of MyST, style guide, and English syntax and grammar fixes.

Would you please take a look at moving Node.js and nvm installation to the top of the page after the introduction, and see if there is something that can be reused from existing installation documentation? Thank you!

petschki committed:
Update docs/classic-ui/theming/barceloneta.md

Co-Authored-By: Steve Piercy <web@stevepiercy.com>

petschki committed:
link to installation page

petschki commented:
@stevepiercy thank you for your review. I've changed the doc and put the Node.js information at the top  with a link to `Install Plone from its packages`. There I've found details on installing python, node, nvm ...

stevepiercy commented on docs/classic-ui/theming/barceloneta.md:
```diff
@@ -11,59 +11,334 @@ myst:
 (classic-ui-theming-barceloneta-theme-package-label)=
 
-## Theme package
+## Create a Classic UI theme addon package
```
```suggestion
## Create a Classic UI theme add-on package
```

stevepiercy committed:
Add a clarifying sentence about the purpose of this chapter.
Break first subsection into two subsections.
s/addon/add-on
MyST and English syntax and grammar fixes.

stevepiercy reviewed (approved):
I decided it would be easier to push a commit 63b366081be99085dae68e3e5f2d7d8208930e51 with my suggestions than go through a formal review. Feel free to change them. Thank you for your contribution. This is super helpful.

petschki commented:
This is good to merge for me 🤗 @gogobd wanted to take a second look over this.
